data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java

```diff
@@ -77,54 +81,83 @@ void setTableKind(WellKnownDataSet tableKind) {
                 }
-                KustoResultColumn col = new KustoResultColumn(jsonCol.getString(COLUMN_NAME_PROPERTY_NAME), columnType, i);
-                columnsAsArray[i] = col;
-                columns.put(jsonCol.getString(COLUMN_NAME_PROPERTY_NAME), col);
             }
         }
 
-        JSONArray exceptions;
-        JSONArray jsonRows = jsonTable.optJSONArray(ROWS_PROPERTY_NAME);
+        ArrayNode exceptions;
+        ArrayNode jsonRows = null;
+        if (jsonTable.has(ROWS_PROPERTY_NAME) && jsonTable.get(ROWS_PROPERTY_NAME).getNodeType() == JsonNodeType.ARRAY) {
+            jsonRows = (ArrayNode) jsonTable.get(ROWS_PROPERTY_NAME);
+        }
         if (jsonRows != null) {
             List<List<Object>> values = new ArrayList<>();
-            for (int i = 0; i < jsonRows.length(); i++) {
-                Object row = jsonRows.get(i);
-                if (row instanceof JSONObject) {
-                    exceptions = ((JSONObject) row).optJSONArray(EXCEPTIONS_PROPERTY_NAME);
+            for (int i = 0; i < jsonRows.size(); i++) {
+                JsonNode row = jsonRows.get(i);
+                if (jsonRows.get(i).getNodeType() == JsonNodeType.OBJECT) {
+                    exceptions = row.has(EXCEPTIONS_PROPERTY_NAME) ? ((ArrayNode) row.get(EXCEPTIONS_PROPERTY_NAME)) : null;
                     if (exceptions != null) {
-                        if (exceptions.length() == 1) {
-                            String message = exceptions.getString(0);
+                        if (exceptions.size() == 1) {
+                            String message = exceptions.get(0).asText();
                             throw new KustoServiceQueryError(message);
                         } else {
                             throw new KustoServiceQueryError(exceptions, false, EXCEPTIONS_MESSAGE);
                         }
                     } else {
-                        throw new KustoServiceQueryError(((JSONObject) row).getJSONArray(
-                                "OneApiErrors"), true, EXCEPTIONS_MESSAGE);
+                        throw new KustoServiceQueryError((ArrayNode) row.get("OneApiErrors"), true, EXCEPTIONS_MESSAGE);
                     }
                 }
-                JSONArray rowAsJsonArray = jsonRows.getJSONArray(i);
+                ArrayNode rowAsJsonArray = (ArrayNode) jsonRows.get(i);
                 List<Object> rowVector = new ArrayList<>();
-                for (int j = 0; j < rowAsJsonArray.length(); ++j) {
-                    Object obj = rowAsJsonArray.get(j);
-                    if (obj == JSONObject.NULL) {
+                for (int j = 0; j < rowAsJsonArray.size(); ++j) {
+                    JsonNode obj = rowAsJsonArray.get(j);
+                    if (obj.isNull()) {
                         rowVector.add(null);
                     } else {
-                        rowVector.add(obj);
+                        switch (rowAsJsonArray.get(j).getNodeType()) {
```

Comment:
Maybe for next version:
Now that the library doesnt handle conversion by itself we should add a flag to this method indicating if the user want us to do this conversion. This is useful for the scenario of Client.executeToJsonResult() which was added for quick response

Reply:
actually i talked to yehezkel about that - this was a feature for Checkpoint only and therefore this might be an overkill